File: hotels.c
#include <stdio.h>

int main() {
    int iniIndex=0;
    int soma = 0;
    int maior = 0;
    int n, m;

    scanf("%d %d", &n, &m);
    int values[n];

    for(int j=0;j<n;j++){
          scanf("%d", &values[j]);
    }

    for (int i=0; i<n; i++){
        if (soma+values[i]<m){
            soma += values[i];
        }
        else if(soma==m){
            maior = m;
        }
        else{
            if(soma>maior){
                maior=soma;
            }   
            while(soma+values[i]>m){
                soma -= values[iniIndex];
                iniIndex += 1;
            }  
            soma += values[i];
            if(soma>maior){
                maior=soma;
            } 
        }  
    }   
    printf("%d\n", maior);

    return 0;
}

File: hotels.py
n, m = [int(x) for x in input().split()]
values = [int(x) for x in input().split()]

iniIndex=0
soma = 0
maior = 0
for i in range(0, len(values)):
    if soma+values[i]<m:
        soma += values[i]
    elif soma==m:
        break
    else:
        if soma>maior:
            maior=soma
        while soma+values[i]>m:
            soma -= values[iniIndex]
            iniIndex += 1
        soma += values[i]
        if soma>maior:
            maior=soma
    
print(maior)